logprep/processor/generic_adder/processor.py

```diff
@@ -43,15 +56,131 @@ def __init__(self, name: str, skipped_fields: List[str]):
+        """
 
-        for dotted_field, value in rule.add.items():
+    rule_class = GenericAdderRule
+    db_table = None
+
+    def __init__(self, name: str, configuration: Processor.Config, logger: Logger):
+        """Initialize a generic adder instance.
+        Performs a basic processor initialization. Furthermore, a SQL database and a SQL table are
+        being initialized if a SQL configuration exists.
+        Parameters
+        ----------
+        name : str
+           Name for the generic adder.
+        configuration : Processor.Config
+           Configuration for SQL adding and rule loading.
+        logger : logging.Logger
+           Logger to use.
+        """
+        super().__init__(name, configuration, logger)
+
+        sql_config = configuration.sql_config
+        self._db_connector = MySQLConnector(sql_config, logger) if sql_config else None
+
+        if GenericAdder.db_table is None:
+            GenericAdder.db_table = self._db_connector.get_data() if self._db_connector else None
+        self._db_table = GenericAdder.db_table
+
+    def _apply_rules(self, event: dict, rule: GenericAdderRule):
+        """Apply a matching generic adder rule to the event.
+         Add fields and values to the event according to the rules it matches for.
+        Additions can come from the rule definition, from a file or from a SQL table.
+        The SQL table is initially loaded from the database and then reloaded if it changes.
+        At first it checks if a SQL table exists and if it will be used. If it does, it adds all
+        values from a matching row in the table to the event. To determine if a row matches, a
+        pattern is used on a defined value of the event to extract a subvalue that is then matched
+        against a value in a defined column of the SQL table. A dotted path prefix can be applied to
+        add the new fields into a shared nested location.
+        If no table exists, fields defined withing the rule itself or in a rule file are being added
+        to the event.
+        Parameters
+        ----------
+        event : dict
+           Name of the event to add keys and values to.
+        rule : GenericAdderRule
+           A matching generic adder rule.
+        Raises
+        ------
+        DuplicationError
+            Raises if an addition would overwrite an existing field or value.
+        """
+
+        if self._db_connector and self._db_connector.check_change():
+            self._db_table = self._db_connector.get_data()
+
+        conflicting_fields = []
+
+        # Either add items from a sql db table or from the rule definition and/or a file
+        if rule.db_target and self._db_table:
+            # Create items to add from a db table
+            items_to_add = []
+            if rule.db_pattern:
+                # Get the sub part of the value from the event using a regex pattern
+                value_to_check_in_db = self._get_dotted_field_value(event, rule.db_target)
```

Comment:
use the get_dotted_field_value from util. this method in base class will disappear in future

Reply:
I've changed it.